helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -338,14 +525,37 @@ public Object removeRoutingTableChangeListener(
     return getInstancesForResource(resourceName, state);
   }
 
+  public Set<InstanceConfig> getInstances(String resourceName, String state,
+      PropertyType propertyType) {
+    return getInstancesForResource(resourceName, state, propertyType);
+  }
+
+  public Set<InstanceConfig> getInstances(String resourceName, String state,
+      PropertyType propertyType, String type) {
+    return getInstancesForResource(resourceName, state, propertyType, type);
+  }
+
   /**
    * returns all instances for {resource} that are in a specific {state}.
    * @param resourceName
    * @param state
    * @return empty list if there is no instance in a given state
    */
   public Set<InstanceConfig> getInstancesForResource(String resourceName, String state) {
-    return _routingTableRef.get().getInstancesForResource(resourceName, state);
+    String key = getRoutingTableKey();
+    return _routingTableRef.get(key).get().getInstancesForResource(resourceName, state);
+  }
+
+  public Set<InstanceConfig> getInstancesForResource(String resourceName, String state,
```

Comment:
Same here. If not in the design, let's hold this for now.

Reply:
removed.